src/util/http/websocket/WebSocketSession.h

```diff
@@ -44,7 +44,9 @@ class WebSocketSession {
   /// The main interface for this class. The HTTP server is supposed to check
   /// if an HTTP request is a websocket upgrade request and delegate further
   /// handling to this method if it is the case. It then accepts the websocket
-  /// connection and "blocks" for the lifetime of it.
+  /// connection and "blocks" for the lifetime of it. Make sure to call this
+  /// coroutine on an explicit strand, and make sure the passed socket's
+  /// executor is that same strand, otherwise there may be race conditions.
```

Comment:
@joka921 We should talk about the reason behind this decision. Currently this is given (but implemented too restrictive, I talked to you about this), but I deliberately removed the creation of the strand in this class because (like many other code snippets in this PR) it seems to imply different behaviour contrary to what it's actually doing.
So by removing explicit code that doesn't work, for now I'll keep the implicit behaviour so future developers won't have to work with false assumptions.

Reply:
Yes, we should talk about why the sessions can't create their own strands.